fix: fail fast on planning list variable sources (#2469)
TimefoldAI/timefold-solver: docs/src/modules/ROOT/pages/domain-modeling/modeling-planning-problems.adoc

|Simple Variable Name
|"variableName"

> |Refers to a variable (genuine or shadow) on the same planning entity. A planning list variable cannot be a source and fails fast, since the shadow variable would not be updated when the list variable or the shadow variables of its elements change.

Reviewer: ```suggestion
|Refers to a variable (genuine or shadow) on the same planning entity. A planning list variable cannot be used as a source.
```
